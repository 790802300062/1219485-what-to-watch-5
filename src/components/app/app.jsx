import React from 'react';
import PropTypes from 'prop-types';
import {
  Switch,
  Route,
  BrowserRouter
} from 'react-router-dom';

import MainPage from '../main-page/main-page';
import Login from '../login/login';
import MyList from '../my-list/my-list';
import Film from '../film/film';
import AddReview from '../add-review/add-review';
import Player from '../player/player';

const App = (props) => {
  const {moviePoster} = props;

  return (
    <BrowserRouter>
      <Switch>
        <Route exact path="/" >
          <MainPage moviePoster={moviePoster} />
        </Route>
        <Route exact path="/login">
          <Login />
        </Route>
        <Route exact path="/mylist">
          <MyList />
        </Route>
        <Route exact path="/films/:id/review">
          <AddReview />
        </Route>
        <Route exact path="/films/:id">
          <Film />
        </Route>
        <Route exact path="/player/:id">
          <Player />
        </Route>
        <Route>
          <h1>404: Not Found</h1>
        </Route>
      </Switch>
    </BrowserRouter>
  );
};

App.propTypes = {
  moviePoster: PropTypes.objectOf({
    title: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    year: PropTypes.number.isRequired
  }).isRequired
};

export default App;
