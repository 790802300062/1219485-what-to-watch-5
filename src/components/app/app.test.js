import React from "react";
import renderer from "react-test-renderer";
import {App} from "./app";
import {noop, movieCard} from "../../test-data";

describe(`App render`, () => {
  it(`renders App component`, () => {
    const tree = renderer
      .create(
          <App
            movieCard={movieCard}
            films={[]}
            isLoading={true}
            loadFilmList={noop}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
