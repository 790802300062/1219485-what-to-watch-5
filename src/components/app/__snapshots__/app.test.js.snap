// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render renders App component 1`] = `
<div>
  Loading...
</div>
`;
